tests/integration/_conftest/transaction_registry.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass, field
+
+from starknet_py.net.client_models import InvokeFunction, StarknetTransaction
+from starknet_py.net.models.transaction import DeployAccount
+
+
+@dataclass
+class TransactionRegistry:
```

Comment:
Why bother with 2 separate classes doing basically the same thing? (capturing the txns)

Reply:
`GatewayClient` is instantiated multiple times (each network command creates a `GatewayClient`). `TransactionRegistry` stores transactions from all gateway clients instantiated within a single test. 

These classes already exist in the code base. This PR splits the `protostar_fixture.py` file.